memclear.c

```diff
@@ -62,26 +78,30 @@ EFI_STATUS efi_main(EFI_HANDLE image_handle, EFI_SYSTEM_TABLE* system_table)
 		if (r != EFI_SUCCESS)
 			goto err;
 
-		r = gBS->GetMemoryMap(&MemoryMapSize, MemoryMap, NULL, &DescriptorSize, NULL);
+		r = gBS->GetMemoryMap(&MemoryMapSize, MemoryMap, &MapKey, &DescriptorSize, NULL);
 	} while (r == EFI_BUFFER_TOO_SMALL);
 
 	asm volatile ("movq %%rsp, %0" : "=r"(rsp));
+	asm volatile ("lea (%%rip), %0" : "=r"(rip));
+	debug_hex(L" rsp: 0x", rsp);
+	debug_hex(L" rip: 0x", rip);
 
+	UINTN ProtectedMemSize = 0;
 	/* Loop through every unoccupied memory range and reserve + clear it. */
 	for (Map = MemoryMap;
-	     Map != ((void*)MemoryMap + MemoryMapSize);
-	     Map = NextMemoryDescriptor(Map, DescriptorSize)) {
+			Map != ((void*)MemoryMap + MemoryMapSize);
+			Map = NextMemoryDescriptor(Map, DescriptorSize)) {
+		UINT64 i;
+		void *addr = (void*)Map->PhysicalStart;
+		UINTN start = Map->PhysicalStart;
+		UINTN end = start + (PAGE_SIZE * Map->NumberOfPages);
+
+		debug_hex(L"\nStart: 0x", start);
+		debug_hex(L"End:   0x", end);
+		debug_hex(L"Size:   0x", end - start);
+		debug_hex(L"Memory Type: ", Map->Type);
+
 		if (Map->Type == EfiConventionalMemory) {
```

Comment:
You're doing a switch() on the memory type just below. Why not combine this if with the switch?